From 59d6029a7102083c125d32ee6e2cff70c3ae2b7e Mon Sep 17 00:00:00 2001
From: Priyajit Ghosh <prghosh@habana.ai>
Date: Wed, 20 Nov 2024 06:07:41 +0200
Subject: [PATCH] [FSW-11114] fs1: adds export.elf & modify makefile

In this patch adds export.elf file and modify
makefile to include export.elf.
Now onwards boot_rom code can be usable from
ppboot and other firmware components via export.elf.

Change-Id: I76965c33e364788fd81f50c49bfa1cf9d5c11410
Signed-off-by: Priyajit Ghosh <prghosh@habana.ai>
---

diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index 86ba7d6..f6c7fda 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -7,11 +7,8 @@
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/common
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/$(DEV_PREFIX)
 
-# TODO: Remove when bootrom exports is available
 ARBOOT_EXPORTS_FILE := $(HL_DEV_ARCH)/exports.elf
-ifeq ($(shell test -e $(ARBOOT_EXPORTS_FILE) && echo -n yes),yes)
 LDFLAGS += -Wl,--just-symbols $(ARBOOT_EXPORTS_FILE)
-endif
 
 PPBOOT_SRCS_C := $(filter-out lib/common/module_i2c_slave.c,$(PPBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(filter-out lib/common/pif_ts.c,$(PPBOOT_SRCS_C))
diff --git a/arch/fs1/arc/exports.elf b/arch/fs1/arc/exports.elf
new file mode 100644
index 0000000..41455c2
--- /dev/null
+++ b/arch/fs1/arc/exports.elf
Binary files differ
